src/AdsImporter/Importer.php

```diff
@@ -0,0 +1,245 @@
+<?php
+
+
+namespace Adshares\AdsManager\AdsImporter;
+
+use Adshares\Ads\AdsClient;
+use Adshares\Ads\Driver\CommandError;
+use Adshares\AdsManager\AdsImporter\Exception\AdsClientException;
+use Adshares\AdsManager\Document\Block;
+use Adshares\AdsManager\Document\Package;
+use Adshares\AdsManager\Document\Node;
+use Adshares\AdsManager\Document\Account;
+use Adshares\Ads\Exception\CommandException;
+use Adshares\AdsManager\AdsImporter\Database\DatabaseMigrationInterface;
+use Adshares\AdsManager\Document\Transaction;
+use Adshares\AdsManager\Helper\NumericalTransformation;
+use Psr\Log\LoggerInterface;
+
+class Importer
+{
+    /**
+     * @var AdsClient
+     */
+    private $client;
+
+    /**
+     * @var DatabaseMigrationInterface
+     */
+    private $databaseMigration;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var int
+     */
+    private $genesisTime;
+
+    /**
+     * @var int
+     */
+    private $blockSeqTime = 32;
+
+    /**
+     * @var ImporterResult
+     */
+    private $importerResult;
+
+    /**
+     * Importer constructor.
+     * @param AdsClient $client
+     * @param DatabaseMigrationInterface $databaseMigration
+     * @param LoggerInterface $logger
+     * @param int $genesisTime
+     * @param int $blockSeqTime
+     */
+    public function __construct(
+        AdsClient $client,
+        DatabaseMigrationInterface $databaseMigration,
+        LoggerInterface $logger,
+        int $genesisTime,
+        int $blockSeqTime
+    ) {
+        $this->client = $client;
+        $this->databaseMigration = $databaseMigration;
+        $this->logger = $logger;
+        $this->genesisTime = $genesisTime;
+        $this->blockSeqTime = $blockSeqTime;
+        $this->importerResult = new ImporterResult();
+    }
+
+    public function import(): ImporterResult
+    {
+        $getMeResponse = $this->client->getMe();
+        $startTime = $this->getStartTime();
+        $endTime = (int)$getMeResponse->getPreviousBlockTime()->format('U');
+
+        $blockId = NumericalTransformation::decToHex($startTime);
+
+        // update nodes
+        $this->updateNodes();
+
+        do {
+            try {
+                $blockResponse = $this->client->getBlock($blockId);
+                /** @var Block $block */
+                $block = $blockResponse->getBlock();
+
+                $blockTransactions = $this->addPackagesFromBlock($block);
+
+                $block->setTransactionCount($blockTransactions);
+                $this->databaseMigration->addBlock($block);
+                ++$this->importerResult->blocks;
+            } catch (CommandException $ex) {
+                if ($ex->getCode() !== CommandError::GET_BLOCK_INFO_UNAVAILABLE) {
+                    $this->addExceptionToLog($ex, 'get_block', ['block' => $blockId]);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "block" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1192)